tests/cases/conformance/types/keyof/keyofAndIndexedAccess.ts

```diff
@@ -31,8 +31,8 @@ type K01 = keyof string;  // "toString" | "charAt" | ...
 type K02 = keyof number;  // "toString" | "toFixed" | "toExponential" | ...
 type K03 = keyof boolean;  // "valueOf"
 type K04 = keyof void;  // never
-type K05 = keyof undefined;  // never
-type K06 = keyof null;  // never
+type K05 = keyof undefined;  // string | number | symbol
+type K06 = keyof null;  // string | number | symbol

```

Comment:
For me, as a user - this doesn't quite make any sense. It's super hard for me to reason about what kind of runtime behavior this is supposed to represent. I've also checked it with this PR and this creates an unsoundness and might lead to runtime errors like here:
```ts
function test<T, K extends keyof T>(t: T, k: K) {
    return t[k]; // oops, runtime error if we allow the call below
}

test(null, 'foo') // OK with this patch, error with 4.7
```

Reply:
Given the new 4.8 behavior, `test` itself should have an error in its implementation due to not having a `{ }` constraint.

Reply:
First experiment wasn't optimal, now deferring types like `keyof (T & {})` such that effects of removing `undefined` and `null` are properly reflected.